src/main/java/uk/gov/register/store/postgres/PostgresDriverTransactional.java

```diff
@@ -46,11 +47,27 @@ protected PostgresDriverTransactional(Handle handle, MemoizationStore memoizatio
     }
 
     public static void useTransaction(DBI dbi, MemoizationStore memoizationStore, Consumer<PostgresDriverTransactional> callback) {
-        dbi.useTransaction(TransactionIsolationLevel.SERIALIZABLE, (handle, status) -> {
-            PostgresDriverTransactional postgresDriver = new PostgresDriverTransactional(handle, memoizationStore);
+        TransactionalMemoizationStore transactionalMemoizationStore = new TransactionalMemoizationStoreImpl(memoizationStore);
+
+        // Get the handle
+        Handle handle = dbi.open();
+
+        try {
+            handle.setTransactionIsolation(TransactionIsolationLevel.SERIALIZABLE);
+            handle.begin();
+
+            PostgresDriverTransactional postgresDriver = new PostgresDriverTransactional(handle, transactionalMemoizationStore);
             callback.accept(postgresDriver);
             postgresDriver.writeStagedData();
-        });
+
+            handle.commit();
+            transactionalMemoizationStore.commitEntries();
```

Comment:
what is going to happen in the exception block with `handle.rollback()` if `transactionalMemoizationStore.commitEntries();` chucks an exception?
maybe the ordering of committing can solve this